include/zephyr/bluetooth/conn.h

```diff
@@ -1358,6 +1358,22 @@ struct bt_conn_auth_info_cb {
  */
 int bt_conn_auth_cb_register(const struct bt_conn_auth_cb *cb);
 
+/** @brief Register authentication callbacks used only for a given connection.
+ *
+ *  Register callbacks to handle authenticated pairing for a given connection.
+ *  The set of callbacks overlays globally defined callbacks.
+ *
+ *  This function can be used only for Bluetooth LE connections.
+ *  The authentication callbacks for a given connection cannot be updated if security procedures in
+ *  the SMP module have already started. This function can be called only once for connection.
```

Comment:
Would it perhaps make sense to mention that the only safe place to avoid such a race is to call this in the `connected` callback. Any point after that the stack can't guarantee that something (e.g. remote device) triggers an authentication procedure.

Also: could you configure your editor to wrap lines at 80 characters. I know the absolute max as per coding style is now days 100, but that should really only be reserved for code formatting when it makes the code more readable. Documentation is at least in my eyes more readable with shorter lines.

Reply:
Changed to 80 characters per line.

Reply:
I would prefer to stick with information that:
```
 *  The authentication callbacks for a given connection cannot be updated if                         
 *  security procedures in the SMP module have already started. This function                        
 *  can be called only once for connection
```
I think it's more generic and I would say that in many use-cases the per-conn callback assignment would be triggered by receiving some data from peer (e.g. Fast Pair auth callbacks are registered while handling GATT write on Fast Pair characteristic - that is the moment when peripheral starts to be aware that it should use a vendor-specific bonding procedure).

Keep in mind that if central wants to follow a vendor-specific procedure, then the procedure must be triggered before SMP security is started (otherwise peripheral has no way to report proper bonding capabilities). Also the worst case scenario is using default authentication callbacks combined with clearly reporting an error code to caller of new API (so there should be unexpected behavior and user will be aware about the issue).